### ISSUE TITLE
PhoneNumber.getCarrierName() throws exception on not found instead of returning null

### ISSUE THREAD

**Issue by georgeabrahamp** [twilio/twilio-java#228]
The various accessors on PhoneNumber throws an IllegalArgumentException when the property is not present. Wouldn't it be better to simply return null in this case?

On the client side I suspect most users transform the Twilio PhoneNumber model into their own and do a copy of needed fields into the user's phone model. When an exception is thrown it needs to be handled before going on to the next field. 

In the snippet below any of the phoneNumber.get* calls could thrown an exception.

```java
       final PhoneNumber phoneNumber = twilioLookupClient.getPhoneNumber(lookupPhoneNumber, countryCode, "carrier".equals(type));

        final PhoneNumberHygieneRecord phoneNumberHygieneRecord = new PhoneNumberHygieneRecord();

        phoneNumberHygieneRecord.setCarrierName(phoneNumber.getCarrierName());
        phoneNumberHygieneRecord.setCountryCode(phoneNumber.getCountryCode());
        phoneNumberHygieneRecord.setEncryptedNationalFormat(phoneNumber.getPhoneNumber());
        phoneNumberHygieneRecord.setEncryptedPhoneNumber(phoneNumber.getFormattedNumber());
        phoneNumberHygieneRecord.setErrorCode(phoneNumber.getErrorCode() == null ? "NONE" :  phoneNumber.getErrorCode().toString());
        phoneNumberHygieneRecord.setMobileCountryCode(phoneNumber.getMobileCountryCode());
        phoneNumberHygieneRecord.setMobileNetworkCode(phoneNumber.getMobileNetworkCode());
        phoneNumberHygieneRecord.setType(phoneNumber.getType() == null ? "NONE" : phoneNumber.getType().name());

        return phoneNumberHygieneRecord;

```

**Comment by jingming** [comment]
@georgeabrahamp We have just released a new version (7.0.0) of the SDK. This version uses appropriate Jackson parsing and will no longer throw exceptions when trying to get properties.

**Comment by georgeabrahamp** [comment]
thx
